packages/block-editor/src/components/link-control/index.js

```diff
@@ -203,29 +205,24 @@ function LinkControl( {
 			wrapperNode.current.ownerDocument.activeElement
 		);
 
-		setSettingsOpen( false );
 		setIsEditingLink( false );
 	};
 
 	const handleSelectSuggestion = ( updatedValue ) => {
-		onChange( {
+		setNewValue( {
 			...updatedValue,
 			title: internalTextInputValue || updatedValue?.title,
 		} );
-		stopEditing();
 	};
 
 	const handleSubmit = () => {
-		if (
-			currentUrlInputValue !== value?.url ||
-			internalTextInputValue !== value?.title
-		) {
-			onChange( {
-				...value,
-				url: currentUrlInputValue,
-				title: internalTextInputValue,
-			} );
-		}
+		const valueToSubmit = {
+			...newValue,
+			url: currentUrlInputValue,
+			title: internalTextInputValue,
+		};
+		setNewValue( valueToSubmit );
+		onChange( valueToSubmit );
```

Comment:
The original code didn't trigger onChange if nothing changed. Is this still true?

Reply:
Thanks for the review 😊 🙏 @draganescu 
When submitted, Here I triggered onChange even the the 'title' & 'url' wasn't changed. I described the following in my PR's description that's I believe related to your question:


> ** 3) in 'handleSubmit' of 'index.js' removed the if check as looks like it's unnecessary. Because if user decide to submit without making any change to 'url' or 'title' there's no harm in still submitting it, also there's other change can happen like 'open in new tab settings' and user may just wanna submit that.

Reply:
Just an update: I haven't found any issue caused by this so far. I'm assuming U haven't too. 
But if U see any `issue` or `potential issue/problem` with this changes please feel free to share. 

I'm not resolving this conversation at this moment. But looks so far it can be resolved if no issue found from your end.
Thanks again. 😊